core/src/main/scala/org/apache/spark/sql/snappyParsers.scala

```diff
@@ -630,14 +633,21 @@ class SnappyParser(caseSensitive: Boolean)
  * Snappy dialect uses a much more optimized parser and adds SnappyParser
```

Comment:
However, I do agree that SnappyContext is not the proper Spark's way of doing this. The SnappyContext is more like a context at execution time whereas query hint is a compilation time context. In other words it should be possible to pick a plan created by some other SnappyContext and run in some other (derby has separate execution and compilation contexts for that reason). Ideally this should be encapsulated in a wrapper LogicalPlan which is then captured in execution plan in our analyser and then picked up where required. This was quite some bit of work so I skipped (the original PR already lead to 7 PRs which are still big).